src/base.jl

```diff
@@ -2619,6 +2619,28 @@ function convert_component!(
     remove_component!(sys, old_load)
 end
 
+"""
+Set the number of a bus.
+"""
+function set_bus_number!(sys::System, bus::ACBus, number::Int)
+    if number in sys.bus_numbers
+        throw(ArgumentError("bus number $number is already stored in the system"))
+    end
+    bus.number = number
+    push!(sys.bus_numbers, number)
```

Comment:
Don't you also have to pop the old number? At least in my use case, `sys.bus_numbers` is populated with all the old bus numbers